Add UNIX lesson

raynamharris opened the pull request:
## PR Checklist

- [x] Release label added
- [x] PR category label added
- [x] PR mergeable
- [x] Clean build logs
- [x] Linked related issues #532 
- [x] Colorblindness test https://www.toptal.com/designers/colorfilter/

## PR Description

Adding UNIX lessons

https://cfde-training-and-engagement--537.com.readthedocs.build/en/537/General-Tools/UNIX/

## Timeline

June release

raynamharris committed:
Update .pages

raynamharris committed:
Create index.md

add hackmd text

raynamharris committed:
Create intro.md

hackmd text

raynamharris committed:
Update intro.md

formatting

raynamharris committed:
Update index.md

raynamharris committed:
Create natigate.md

formatting

raynamharris committed:
Create .pages

raynamharris committed:
Rename natigate.md to navigate.md

raynamharris committed:
Merge branch 'dev' into add-unix

jeremywalter committed:
Update navigate.md

raynamharris committed:
Update index.md

formatting

raynamharris committed:
Update navigate.md

formatting

raynamharris committed:
Update navigate.md

raynamharris committed:
Update intro.md

header levels

raynamharris committed:
Create read.md

raynamharris committed:
Create work.md

formatting

raynamharris committed:
Create find.md

raynamharris committed:
Create redirect.md

hmm... doesn't actually cover > and >>

raynamharris committed:
Update redirect.md

add append

raynamharris committed:
Update index.md

add resource

raynamharris committed:
Create appendix.md

raynamharris committed:
Update .pages

raynamharris committed:
Update .pages

raynamharris committed:
Update index.md

rm launch binder button

jeremywalter committed:
rearrange general tools (Conda/Snakemake/UNIX first)

jeremywalter committed:
Add UNIX

jeremywalter committed:
update ls output examples

jeremywalter committed:
re-format challenges

jeremywalter committed:
update head output, gunzip text file name

jeremywalter committed:
fix success

jeremywalter committed:
fix head, remove success, fix UNIX commands

raynamharris committed:
Update read.md

key point formatting

raynamharris commented:
thanks @jeremywalter !

jeremywalter committed:
fix mothur name

jeremywalter committed:
remove extra full stop

jeremywalter committed:
add mybinder hyperlink to key points

jeremywalter committed:
update ls output, change format for Input/Output display

jeremywalter committed:
remove more, add tail, update key points

jeremywalter committed:
remove script mention (file not here and no context)

jeremywalter committed:
spelling fix

jeremywalter committed:
link fix, challenge formatting

raynamharris commented:
closes #539

raynamharris commented:
will likely have conflicts with #535 with the .pages file due to both lessons needing to be added to the nav bar. i plan on merging this one first and fixing the conflict in the other second.

jeremywalter committed:
formatting for books titles in intro

jeremywalter committed:
formatting for commands and accessibility (remove contraction)

jeremywalter committed:
update outputs and fix formatting for inputs and outputs

jeremywalter committed:
fix last input/output

jeremywalter committed:
fix special format navigate

jeremywalter committed:
add tail references and gunzip after text

raynamharris committed:
Merge branch 'dev' into add-unix

raynamharris committed:
Update work.md

fix warning

raynamharris commented:
thanks jeremy. excellent updates. ready for approval

jessicalumian commented on docs/General-Tools/UNIX/find.md:
```diff
@@ -0,0 +1,176 @@
+
+A lot of the time we want to know if a file contains what we expect. A useful thing to do is to be able to **search the contents of files** for a particular string of characters you would like to find.  We can use the file pattern searcher `grep` to find things.
+
+The `MiSeq/` directory contains many of the sequence files ending in`.fastq`. We expect these files to contain information in a particular format throughout the file with four lines of information for each sequence string. Looking through a million line file using less will take a long time. Rather than manually looking at the whole file, we can print only a portion of the file's contents to stan ...
```
```suggestion
The `MiSeq/` directory contains many of the sequence files ending in`.fastq`. We expect these files to contain information in a particular format throughout the file with four lines of information for each sequence string. Looking through a million-line file using less will take a long time. Rather than manually looking at the whole file, we can print only a portion of the file's contents to standard output. 
```

jessicalumian commented on docs/General-Tools/UNIX/find.md:
```diff
@@ -0,0 +1,176 @@
+
+The `MiSeq/` directory contains many of the sequence files ending in`.fastq`. We expect these files to contain information in a particular format throughout the file with four lines of information for each sequence string. Looking through a million line file using less will take a long time. Rather than manually looking at the whole file, we can print only a portion of the file's contents to stan ...
+
+Let's say you'd like to find the sequence `CATTAG` in your MiSeq files. We can use function `grep` to search for  `CATTAG` in one or all all of the fastq files located in our current working directory.
```
```suggestion
Let's say you'd like to find the sequence `CATTAG` in your MiSeq files. We can use the function `grep` to search for  `CATTAG` in one or all of the fastq files located in our current working directory.
```

jessicalumian commented on docs/General-Tools/UNIX/navigate.md:
```diff
@@ -0,0 +1,203 @@
+
+### `cd`
+
+Now we have seen how to list the contents of folders on our computers and what is located in the directory we are presently in. But some of the beauty of the shell is that we can perform activities in locations that we are not currently in. To do this we can either use an absolute path or a relative path. A **relative path** is the path to another directory from the one you are currently in. An * ...
```
```suggestion
Now we have seen how to list the contents of folders on our computers and what is located in the directory we are presently in. But some of the beauty of the shell is that we can perform activities in locations that we are not currently in. To do this we can either use an absolute path or a relative path. A **relative path** is the path to another directory from the one you are currently in. An **absolute path** starts from the root and ends in the appropriate subdirectory. 
```

jessicalumian commented on docs/General-Tools/UNIX/read.md:
```diff
@@ -0,0 +1,87 @@
+
+Now that we know what files exist on our computer, it's time to look at the contents of the file. There are multiple ways to look at the contents of a file. 
+
+The `cat` command prints the entirety of a file to the stdout of our computer. We can scroll through files using the `less` command. Less is a safe way of looking at the contents of a file without the ability to change it. `head` prints, by default the first 10 lines of a file and `tail` prints the last 10 lines.
```
```suggestion
The `cat` command prints the entirety of a file to the stdout of our computer. We can scroll through files using the `less` command. Less is a safe way of looking at the contents of a file without the ability to change it. `head` prints, by default, the first 10 lines of a file and `tail` prints the last 10 lines.
```

jessicalumian commented on docs/General-Tools/UNIX/redirect.md:
```diff
@@ -0,0 +1,100 @@
+
+### `|`
+
+Let's pipe (`|`) the output of a grep search to the command head to only show the first 10 lines. This is very handy when you only want to view snapshot of the data. 
```
```suggestion
Let's pipe (`|`) the output of a grep search to the command head to only show the first 10 lines. This is very handy when you only want to view a snapshot of the data. 
```

jessicalumian commented on docs/General-Tools/UNIX/redirect.md:
```diff
@@ -0,0 +1,100 @@
+
+### For loops
+
+However, the two lines above told us, how many times CATTAG appeared in all of the .fastq files in total. 
```
```suggestion
However, the two lines above tell us how many times CATTAG appeared in all of the .fastq files in total. 
```

jessicalumian commented on docs/General-Tools/UNIX/redirect.md:
````diff
@@ -0,0 +1,100 @@
+done
+```
+
+If we cant to know which file these results came from, we can add an echo statement to echo the variable.
````
```suggestion
If we want to know which file these results came from, we can add an echo statement to echo the variable.
```

jessicalumian commented on docs/General-Tools/UNIX/redirect.md:
```diff
@@ -0,0 +1,100 @@
+
+### `>`
+
+You can redirect the output from the screen to a file using `>`. Note that `>` will overright existing conent, but `>>` will append. 
```
```suggestion
You can redirect the output from the screen to a file using `>`. Note that `>` will overwrite existing content, but `>>` will append. 
```

jessicalumian commented on docs/General-Tools/UNIX/work.md:
````diff
@@ -0,0 +1,76 @@
+mkdir data results images/
+mkdir -p data/results/images
+```
+The first line creates two directories, results and images. The second line also create two directories, but they are nested with the parent directory data. 
````
```suggestion
The first line creates two directories, results and images. The second line also creates two directories, but they are nested with the parent directory data. 
```

jessicalumian commented on docs/General-Tools/UNIX/work.md:
```diff
@@ -0,0 +1,76 @@
+### `rmdir` 
+
+If you created some files or directories that you do not want, you can remove them with the `rm` and `rmdir` commands. 
+`rmdir` will only remove empty directories, but `rm -r` will remove recursively.
```
```suggestion
`rmdir` will only remove empty directories, but `rm -r` will remove them recursively.
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
````diff
@@ -0,0 +1,112 @@
+
+```bash
+clear
+pwd
````
```suggestion
PS1='$ '
clear
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
````diff
@@ -0,0 +1,112 @@
+```bash
+clear
+pwd
+PS1="\w $ "
````
```suggestion

```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+ls
+ls --help
+ls -l books
```
```suggestion
cd books/
pwd
ls -lh
```

raynamharris replied:
keeping as is

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+ls
+ls --help
+ls -l books
+ls -l MiSeq
```
```suggestion

```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+ls --help
+ls -l books
+ls -l MiSeq
+ls -lh MiSeq
```
```suggestion

```

raynamharris replied:
keeping as is

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+ls -l books
+ls -l MiSeq
+ls -lh MiSeq
+ls -lh *
```
```suggestion

```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+ls -lh MiSeq
+ls -lh *
+
+cd books
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+ls -lh *
+
+cd books
+pwd
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+cd books
+pwd
+ls
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+cd books
+pwd
+ls
+cd ../Miseq
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+pwd
+ls
+cd ../Miseq
+pwd
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+ls
+cd ../Miseq
+pwd
+ls
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+cd ../Miseq
+pwd
+ls
+ls *fastq
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+pwd
+ls
+ls *fastq
+ls F3*
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+ls *fastq
+ls F3*
+
+cd ~books
```
```suggestion
cd ~/books/
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+ls F3*
+
+cd ~books
+
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+cat book.txt
+less book.txt
+
+head *txt
```
```suggestion
head *.txt
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+head *txt
+
+wc -l *txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+head *txt
+
+wc -l *txt
+wc -l *txt | sort
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+wc -l *txt
+wc -l *txt | sort
+wc -l *txt | sort -nr
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+wc -l *txt | sort -nr
+
+gunzip TheWonderfulWizardofOz.txt
+wc -l *txt | sort -nr
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+gunzip TheWonderfulWizardofOz.txt
+wc -l *txt | sort -nr
+
```
```suggestion
cp book.txt book-copy.txt
ls

mv book-copy.txt book-2cities.txt

mkdir data results images/
mkdir -p data/results/images

rmdir data/results/images
rmdir data/results

```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+gunzip TheWonderfulWizardofOz.txt
+wc -l *txt | sort -nr
+
+rm book.txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+wc -l *txt | sort -nr
+
+rm book.txt
+mkdir results
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+rm book.txt
+mkdir results
+
+wc -l *txt | sort -nr > results/book_lengths.txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+mkdir results
+
+wc -l *txt | sort -nr > results/book_lengths.txt
+cat results/book_lengths.txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+wc -l *txt | sort -nr > results/book_lengths.txt
+cat results/book_lengths.txt
+
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+wc -l *txt | sort -nr > results/book_lengths.txt
+cat results/book_lengths.txt
+
+grep "Chapter" *txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+cat results/book_lengths.txt
+
+grep "Chapter" *txt
+grep -i "Chapter" *txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+grep "Chapter" *txt
+grep -i "Chapter" *txt
+grep "CHAPTER" *txt > results/chapter_titles.txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+grep "Chapter" *txt
+grep -i "Chapter" *txt
+grep "CHAPTER" *txt > results/chapter_titles.txt
+cat results/chapter_titles.txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+grep -i "Chapter" *txt
+grep "CHAPTER" *txt > results/chapter_titles.txt
+cat results/chapter_titles.txt
+
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+grep "CHAPTER" *txt > results/chapter_titles.txt
+cat results/chapter_titles.txt
+
+grep "The" *txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+cat results/chapter_titles.txt
+
+grep "The" *txt
+grep "^The" *txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+grep "The" *txt
+grep "^The" *txt
+grep -w "^The" *txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+grep "The" *txt
+grep "^The" *txt
+grep -w "^The" *txt
+grep -w -A 1 "^The" *txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+grep "^The" *txt
+grep -w "^The" *txt
+grep -w -A 1 "^The" *txt
+grep -w  "^The" *txt > results/The.txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+grep -w "^The" *txt
+grep -w -A 1 "^The" *txt
+grep -w  "^The" *txt > results/The.txt
+cat results/The.txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+grep -w -A 1 "^The" *txt
+grep -w  "^The" *txt > results/The.txt
+cat results/The.txt
+
```
```suggestion
cd ~/MiSeq
ls

```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+grep -w  "^The" *txt > results/The.txt
+cat results/The.txt
+
+head README.md
```
```suggestion

```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+head README.md
+head -4 F3D0_S188_L001_R1_001.fastq
+tail -4 F3D0_S188_L001_R1_001.fastq
+less F3D0_S188_L001_R1_001.fastq
```
```suggestion
head -4 HMP_MOCK.v35.fasta 
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+tail -4 F3D0_S188_L001_R1_001.fastq
+less F3D0_S188_L001_R1_001.fastq
+
+wc README.md
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+less F3D0_S188_L001_R1_001.fastq
+
+wc README.md
+wc F3D0_S188_L001_R1_001.fastq
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+wc README.md
+wc F3D0_S188_L001_R1_001.fastq
+wc -l F3D0_S188_L001_R1_001.fastq	
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+wc README.md
+wc F3D0_S188_L001_R1_001.fastq
+wc -l F3D0_S188_L001_R1_001.fastq	
+wc -l *fastq
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+wc F3D0_S188_L001_R1_001.fastq
+wc -l F3D0_S188_L001_R1_001.fastq	
+wc -l *fastq
+wc -l *R1*fastq
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+wc -l F3D0_S188_L001_R1_001.fastq	
+wc -l *fastq
+wc -l *R1*fastq
+wc -l *R1*fastq | sort -nr
```
```suggestion
cd ../data/MiSeq/
grep CATTAG F3D0_S188_L001_R2_001.fastq
grep CATTAG *.fastq
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+wc -l *R1*fastq
+wc -l *R1*fastq | sort -nr
+
+head -4 F3D0_S188_L001_R1_001.fastq
```
```suggestion
wc F3D0_S188_L001_R2_001.fastq
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+wc -l *R1*fastq | sort -nr
+
+head -4 F3D0_S188_L001_R1_001.fastq
+grep "^@M" F3D0_S188_L001_R1_001.fastq
```
```suggestion
wc -l F3D0_S188_L001_R2_001.fastq

```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+head -4 F3D0_S188_L001_R1_001.fastq
+grep "^@M" F3D0_S188_L001_R1_001.fastq
+grep "^@M" F3D0_S188_L001_R1_001.fastq | wc -l
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+head -4 F3D0_S188_L001_R1_001.fastq
+grep "^@M" F3D0_S188_L001_R1_001.fastq
+grep "^@M" F3D0_S188_L001_R1_001.fastq | wc -l
+grep "^@M" *R1*.fastq | wc -l
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+grep "^@M" F3D0_S188_L001_R1_001.fastq
+grep "^@M" F3D0_S188_L001_R1_001.fastq | wc -l
+grep "^@M" *R1*.fastq | wc -l
+
```
```suggestion
mkdir results
grep CATTAG *.fastq > results/files-with-CATTAG.txt

grep CATTAG *.fastq 
grep CATTAG *.fastq | wc -l
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+grep "^@M" F3D0_S188_L001_R1_001.fastq | wc -l
+grep "^@M" *R1*.fastq | wc -l
+
+for file in *R1*.fastq
```
```suggestion
for file in *fastq
do
grep CATTAG $file | wc -l
done
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+grep "^@M" *R1*.fastq | wc -l
+
+for file in *R1*.fastq
+do
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+for file in *R1*.fastq
+do
+echo $file
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+for file in *R1*.fastq
+do
+echo $file
+grep "^@M" $file | wc -l
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+do
+echo $file
+grep "^@M" $file | wc -l
+done
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+mkdir results
+
+
```
```suggestion
history

history > ~/history-2021-nov-17.txt
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+mkdir results
+
+
+for file in *R1*.fastq
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+
+for file in *R1*.fastq
+do
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+for file in *R1*.fastq
+do
+echo $file >> results/read_count.txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+for file in *R1*.fastq
+do
+echo $file >> results/read_count.txt
+grep "^@M" $file | wc -l >> results/read_count.txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+do
+echo $file >> results/read_count.txt
+grep "^@M" $file | wc -l >> results/read_count.txt
+done
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+echo $file >> results/read_count.txt
+grep "^@M" $file | wc -l >> results/read_count.txt
+done
+
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+grep "^@M" $file | wc -l >> results/read_count.txt
+done
+
+head results/read_count.txt
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+done
+
+head results/read_count.txt
+
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+head results/read_count.txt
+
+for file in *R1*.fastq
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+head results/read_count.txt
+
+for file in *R1*.fastq
+do
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+for file in *R1*.fastq
+do
+echo $file >> results/samples.csv
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+for file in *R1*.fastq
+do
+echo $file >> results/samples.csv
+grep "^@M" $file | wc -l >> results/count.csv
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+do
+echo $file >> results/samples.csv
+grep "^@M" $file | wc -l >> results/count.csv
+paste -d , results/samples.csv results/count.csv > results/read_count.csv
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+echo $file >> results/samples.csv
+grep "^@M" $file | wc -l >> results/count.csv
+paste -d , results/samples.csv results/count.csv > results/read_count.csv
+done
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+grep "^@M" $file | wc -l >> results/count.csv
+paste -d , results/samples.csv results/count.csv > results/read_count.csv
+done
+
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+paste -d , results/samples.csv results/count.csv > results/read_count.csv
+done
+
+head results/read_count.csv
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+done
+
+head results/read_count.csv
+
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+head results/read_count.csv
+
+cd
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+head results/read_count.csv
+
+cd
+history
```
```suggestion
```

jessicalumian commented on docs/General-Tools/UNIX/appendix.md:
```diff
@@ -0,0 +1,112 @@
+
+cd
+history
+history > history.txt
```
```suggestion
```

jessicalumian reviewed (changes requested):
I made typo fixes, and the biggest thing was changing the code in the appendix.md file. I ran through the lesson code in an RStudio binder and it all worked. The code in the appendix.md file did not match what was in the lesson, so I made a bunch of suggestions to make that code match.

Happy to change this to "approve" now if there's a logistical reason. Otherwise I'll set this to "Request Changes" and "Approve" later.

raynamharris committed:
Update docs/General-Tools/UNIX/find.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/find.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/navigate.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/read.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/redirect.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/appendix.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/appendix.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/appendix.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/appendix.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/appendix.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris commented:
awesome! thanks @jessicalumian changes incorporated

raynamharris committed:
Update docs/General-Tools/UNIX/redirect.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/redirect.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/appendix.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/appendix.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/appendix.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update docs/General-Tools/UNIX/appendix.md

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Apply suggestions from code review

Co-authored-by: jessicalumian <jessica.mizzi@gmail.com>

raynamharris committed:
Update redirect.md

rename history file

raynamharris committed:
Update appendix.md

update history file name

jessicalumian committed:
updated appendix to match current lesson content

raynamharris committed:
Update intro.md

raynamharris committed:
Update work.md

remove or

raynamharris committed:
Update find.md

raynamharris committed:
Update UNIX.md

first pass syncing hackmd source file with lesson material

jeremywalter commented:
couple last minute fixes coming (on the latter pages)... please hold...

jeremywalter committed:
update work -- parentheses and explain recursive

raynamharris committed:
Update redirect.md

raynamharris committed:
Update UNIX.md

upate redirect section

raynamharris committed:
Update find.md

jeremywalter committed:
spelling and format updates to find

raynamharris committed:
Update redirect.md

raynamharris committed:
Update appendix.md

jeremywalter committed:
spelling, attempt to fix challenge

raynamharris committed:
Update redirect.md

fix indent

jeremywalter committed:
fix grep challenge (find)

jeremywalter committed:
hint wording for ebook challenge

raynamharris committed:
Update redirect.md

raynamharris committed:
Update appendix.md

raynamharris committed:
Update appendix.md

raynamharris committed:
Update UNIX.md

sync with https://github.com/nih-cfde/training-and-engagement/pull/537/commits/9cba7d86907540266fd22b71ec7015378a1adf14

raynamharris committed:
Merge branch 'dev' into add-unix

raynamharris committed:
Update index.md

add overview

raynamharris committed:
Merge branch 'dev' into add-unix

raynamharris committed:
Update find.md

fix formatting

raynamharris committed:
Update index.md

add prereqs

raynamharris committed:
Update index.md

formatting

raynamharris committed:
Update index.md

links to more resources

raynamharris commented:
looks good :)

![Screen Shot 2022-06-23 at 10 13 04 AM](https://user-images.githubusercontent.com/7544197/175356471-cd5bb684-cda0-4f4e-96b9-44075a9bdb5c.png)
